easydigitaldownloads/easy-digital-downloads
# Update discounts API query to use discount objects

Fixes #8708

Proposed Changes:
1. Update the `get_discounts` method for the API to use the new discount objects.
2. Update the query parameters to correctly handle pagination.

To test:
To retrieve an array of discounts:

```
https://3dd.local/edd-api/discounts/?key=API_KEY&token=API_TOKEN
```

Add `&page=2` to get a second page of discounts (you should get new discounts; with 3.0, I did not).

To retrieve one discount:

Add `&discount=DISCOUNT_ID` to the URL and comment out the line indicated in the issue.

Noting that I opted to use the discount object properties directly instead of the helper functions because the general query was noticeably faster (1.8s to 0.03s).
